common/models/user.js

```diff
@@ -592,18 +609,30 @@ module.exports = function(User) {
   User.hashPassword = function(plain) {
     this.validatePassword(plain);
     var salt = bcrypt.genSaltSync(this.settings.saltWorkFactor || SALT_WORK_FACTOR);
-    return bcrypt.hashSync(plain, salt);
+    var hash = bcrypt.hashSync(plain, salt);
+    if (/^\$2a\$/.test(hash)) {
+      // change the prefix to indicate that we have verified the plain password
+      // is at most 72 chars long
+      hash = '$72$'.concat(hash.slice(4));
```

Comment:
@raymondfeng I see you commented

> >  Make the breaking change, assume there are no users with so long passwords.

> IMO, this assumption is reasonable.

I personally prefer to be careful about assumptions when making breaking changes, but I don't have any data points to objectively decide here.

Considering that both you @raymondfeng and @superkhau think it's ok to make the change breaking, I'll bow to the majority, so to speak :)

In that case we can revert  bfe674c completely (remove it from this patch) and call this discussion closed.